tests/integration_tests/test_wandb.py

```diff
@@ -31,79 +32,165 @@ def _multiobjective_func(trial: optuna.trial.Trial) -> Tuple[float, float]:
 @mock.patch("optuna.integration.wandb.wandb")
 def test_run_initialized(wandb: mock.MagicMock) -> None:
 
+    wandb.sdk.wandb_run.Run = mock.MagicMock
+
     wandb_kwargs = {
         "project": "optuna",
         "group": "summary",
         "job_type": "logging",
         "mode": "offline",
+        "tags": ["test-tag"],
     }
 
-    WeightsAndBiasesCallback(
-        metric_name="mse",
-        wandb_kwargs=wandb_kwargs,
+    WeightsAndBiasesCallback(metric_name="mse", wandb_kwargs=wandb_kwargs, as_multirun=False)
+    wandb.init.assert_called_once_with(
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
+    )
+
+    wandbc = WeightsAndBiasesCallback(
+        metric_name="mse", wandb_kwargs=wandb_kwargs, as_multirun=True
+    )
+    wandb.run = None
+
+    study = optuna.create_study(direction="minimize")
+    _wrapped_func = wandbc.track_in_wandb(_objective_func)
+    study.optimize(_wrapped_func, n_trials=1, callbacks=[wandbc])
+
+    wandb.init = mock.MagicMock()
+    study_id = study._study_id
+    frozen_trial = study.trials[0]
+    trial_id = study._storage.get_trial_id_from_study_id_trial_number(
+        study_id, frozen_trial.number
     )
+    trial = optuna.Trial(study, trial_id)
+
+    _ = _wrapped_func(trial)
 
     wandb.init.assert_called_once_with(
-        project="optuna",
-        group="summary",
-        job_type="logging",
-        mode="offline",
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
+    )
+
+    wandb.init = mock.MagicMock()
+    study.optimize(_objective_func, n_trials=1, callbacks=[wandbc])
+    wandb.init.assert_called_once_with(
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
     )
 
 
 @mock.patch("optuna.integration.wandb.wandb")
 def test_attributes_set_on_epoch(wandb: mock.MagicMock) -> None:
 
-    wandb.config.update = mock.MagicMock()
+    # Vanilla update
+    wandb.sdk.wandb_run.Run = mock.MagicMock
 
+    study = optuna.create_study(direction="minimize", study_name="test_study")
     wandbc = WeightsAndBiasesCallback()
-    study = optuna.create_study(direction="minimize")
     study.optimize(_objective_func, n_trials=1, callbacks=[wandbc])
 
     expected = {"direction": ["MINIMIZE"]}
-    wandb.config.update.assert_called_once_with(expected)
+    wandb.run.config.update.call_args[0][0].pop("x")
+    wandb.run.config.update.call_args[0][0].pop("y")
+    wandb.run.config.update.assert_called_once_with(expected)
```

Comment:
I think we could use [enqueue_trial](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.Study.html#optuna.study.Study.enqueue_trial) here to control parameters suggested, so we can test if `run.config` contains all expected values.
```python
study = optuna.create_study(direction="minimize", study_name="test_study")
study.enqueue_trial({"x": 5.0, "y": 1.0})
study.optimize(_objective_func, n_trials=1, callbacks=[wandbc])
expected = {"direction": ["MINIMIZE"], "x": 5.0, "y": 1.0]}
wandb.run.config.update.assert_called_once_with(expected)
```

Reply:
Fixed. Thanks for pointing it out.